README.md

```diff
@@ -539,30 +524,271 @@ services.Configure<TargetingEvaluationOptions>(options =>
 }
 ```
 
 To use an implementation of `IFeatureDefinitionProvider` it must be added into the service collection before adding feature management. The following example adds an implementation of `IFeatureDefinitionProvider` named `InMemoryFeatureDefinitionProvider`.
 
-```
+``` C#
 services.AddSingleton<IFeatureDefinitionProvider, InMemoryFeatureDefinitionProvider>()
         .AddFeatureManagement()
 ```
 
+It is also possible to provide custom dynamic feature definitions. This is done by implementing the `IDynamicFeatureDefinitionProvider` interface and registering it as mentioned above.
+
+## Dynamic Features
+
+When new features are being added to an application there may come a time when a feature has multiple different proposed design options. The different options for the design of a feature can be referred to as variants of the feature, and the feature itself can be referred to as a dynamic feature. A dynamic feature is a feature that can have different values (variants) extending beyond a simple on/off flag. A common pattern when rolling out dynamic features is to surface the different variants of a feature to different segments of a user base and to see how each variant is perceived. The most well received variant could be the one that gets rolled out to the entire user base, or if necessary the feature could be scrapped. There could be other reasons to expose different variants of a feature, for example using a different version every day of the week. The goal of this method is to establish a model that can help solve these common patterns that occur when rolling out features that can have different variants.
+
+### Consumption
+
+Dynamic features are accessible through the `IDynamicFeatureManager` interface.
+
+``` C#
+public interface IDynamicFeatureManager
+{
+    IAsyncEnumerable<string> GetDynamicFeatureNamesAsync(CancellationToken cancellationToken = default);
+
+    ValueTask<T> GetVariantAsync<T>(string dynamicFeature, CancellationToken cancellationToken = default);
+
+    ValueTask<T> GetVariantAsync<T, TContext>(string dynamicFeature, TContext context, CancellationToken cancellationToken = default);
+}
+```
+
+The dynamic feature manager performs a resolution process that takes the name of a feature and returns a strongly typed value to represent the variant's value.
+
+The following steps are performed during the retrieval of a dynamic feature's variant
+1. Lookup the configuration of the specified dynamic feature to find the registered variants
+2. Assign one of the registered variants to be used.
+3. Resolve typed value based off of the assigned variant.
+
+The dynamic feature manager is made available by using the `AddFeatureManagement` call detailed in the [service registration](./README.md#Service-Registration) section. Make sure to add any required feature variant assigners referenced by dynamic features within the application by using `AddFeatureVariantAssigner`.

```

Comment:
Since you have the sample code below, we don't need to reference the "service registration" section under feature flags.